src/functions/keycloak/update-user-attributes/handler.ts

```diff
@@ -5,76 +5,78 @@ import { CreateSeveraApiService } from "src/services/severa-api-service";
 
 /**
  * Lambda handler to update a user's attributes
- * 
- * @param event API Gateway event 
+ *
+ * @param event API Gateway event
  * @returns Response message as JSON string
  */
-const updateUserAttributeHandler: APIGatewayProxyHandler = async (
-  event: APIGatewayProxyEvent
-  ) => { 
+const updateUserAttributeHandler: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent) => {
   try {
-  if (!event.body) {
-    throw new Error("Request body is missing.");
-  }
+    if (!event.body) {
+      throw new Error("Request body is missing.");
+    }
 
-  const body = JSON.parse(JSON.stringify(event.body));
-  const { id, attributes } = body;
-  const email = attributes?.email; 
+    const body = JSON.parse(JSON.stringify(event.body));
+    const { id, attributes } = body;
+    const email = attributes?.email;
+    const attributeName = event.pathParameters?.attributeName;
+    const allowedKeys = ["isSeveraOptIn"];
 
-  const allowedKeys = ["isSeveraOptIn"]
+    if (!id || !attributes || typeof attributes !== "object") {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ message: "Missing or invalid parameters: 'id' or 'attribute'." })
+      };
+    }
 
-  if (!id || !attributes || typeof attributes !== "object") {
-    return {
-      statusCode: 400,
-      body: JSON.stringify({ message: "Missing or invalid parameters: 'id' or 'attribute'." }),
-    };
-  }
+    const keys = Object.keys(attributes);
+    if (!keys.every((key) => allowedKeys.includes(key))) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ message: "Attributes contain invalid keys." })
+      };
+    }
 
-  const keys = Object.keys(attributes);
-  if (!keys.every(key => allowedKeys.includes(key))) {
-    return {
-      statusCode: 400,
-      body: JSON.stringify({ message: "Attributes contain invalid keys." }),
-    };
-  }
+    const api = CreateKeycloakApiService();
+    const severaApi = CreateSeveraApiService();
 
-  const api = CreateKeycloakApiService();
-  const severaApi = CreateSeveraApiService()
+    if (!Object.prototype.hasOwnProperty.call(attributes, attributeName)) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ message: `Attribute ${attributeName} is missing in request body.` })
+      };
+    }
```

Comment:
I'm pretty sure this is redundant and can be removed as you have this earlier code which is checking the attributes are allowed key:
 const keys = Object.keys(attributes);
    if (!keys.every((key) => allowedKeys.includes(key))) {
      return {
        statusCode: 400,
        body: JSON.stringify({ message: "Attributes contain invalid keys." })
      };
    }